sdk/hybridconnectivity/Azure.ResourceManager.HybridConnectivity/src/Generated/EndpointResource.cs

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.HybridConnectivity.Models;
+
+namespace Azure.ResourceManager.HybridConnectivity
+{
+    /// <summary> A Class representing a EndpointResource along with the instance operations that can be performed on it. </summary>
+    public partial class EndpointResource : ArmResource
+    {
+        /// <summary> Generate the resource identifier of a <see cref="EndpointResource"/> instance. </summary>
+        public static ResourceIdentifier CreateResourceIdentifier(string resourceUri, string endpointName)
+        {
+            var resourceId = $"{resourceUri}/providers/Microsoft.HybridConnectivity/endpoints/{endpointName}";
+            return new ResourceIdentifier(resourceId);
+        }
```

Comment:
We should rename the parameter `resourceUri` to `resourceId` or `scope` since this is actually a resource Id of another resource instead of a URI

Reply:
Another question for this parameter is that do we have any knowledge about what kind of resource can be the scope of this resource? Is there a limited list of valid resource types? Or this resource could apply to any resource type, like subscription, resource group, virtual machine, virtual network, etc?

Reply:
@archerzz can you help us to understand how does codegen generate this method which is not same as other RPs

Reply:
Here `resourceUri` is a real `uri` so change it to `id` is not current

Reply:
I think maybe `scope` is better so we can keep align with `Deployment`.

Reply:
`scope` is right, but can't fix by use `directive`

Reply:
will try later